### ISSUE TITLE
# Question: MODEL.ROI_HEADS.NUM_CLASSES assignment based on the number of classes.

### ISSUE THREAD

**Issue by nick0tine** [pieterblok/maskal#15]
When looking at the implementation I found that `cfg.MODEL.ROI_HEADS.NUM_CLASSES` is assigned as len(config['classes']).

So If there's only a single class then the above param will be assigned to 1.

Curious is, generally we provide `NUM_CLASSES` as 'no of classses' + 1 (for the background) in MASK-RCNN.

Then why the Maskal have above implementation? does here num_classes should be understood as _foreground classes_ only and not _background classes_ ?

Thank you

**Comment by pieterblok** [comment]
@nick0tine Thanks for this question. I'm curious where you got the default example of using classes+1 (including background) from? If I look at the Detectron2 balloon example in Facebook's Colab, they also use Num_Classes=1, not 2.

**Comment by nick0tine** [comment]
Thanks for the clarification,
I did look into detectron2 codebase and the configuration mentioned it should be only foreground classes and 'not background classes' ( there's a closed issue in detectron2 repo regarding same doubt stating the same).

I got it confused, as when not using detectron2; for background we add +1.

My bad, i should have checked on detectron2 docs.
Anyways, thanks for the response.
